ApplicationLibCode/FileInterface/RifRoffFileTools.cpp

```diff
@@ -478,21 +478,101 @@ void RifRoffFileTools::convertToReservoirIndexOrder( int                      nx
         {
             for ( int i = 0; i < nx; i++ )
             {
-                int inIdx         = i * ny * nz + j * nz + k;
+                int inIdx         = i * ny * nz + j * nz + ( nz - k - 1 );
                 activeOut[outIdx] = static_cast<int>( activeIn[inIdx] );
                 outIdx++;
             }
         }
     }
 }
 
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RifRoffFileTools::convertToReservoirIndexOrder( int                       nx,
```

Comment:
Find a way to avoid duplication here. Consider creating and index map first

```
std::vector<int> createIndexMapper( int nx, int ny, int nz )
{
    std::vector<int> mapper;
    mapper.reserve( static_cast<size_t>( nx * ny * nz ) );

    int outIdx = 0;
    for ( int k = 0; k < nz; k++ )
    {
        for ( int j = 0; j < ny; j++ )
        {
            for ( int i = 0; i < nx; i++ )
            {
                int inIdx      = i * ny * nz + j * nz + ( nz - k - 1 );
                mapper[outIdx] = inIdx;
                outIdx++;
            }
        }
    }
    return mapper;
}
```

Reply:
Added a issue for it here: #9601. Will handle it next year!